src/components/date-picker/AccessiblePikaday.ts

```diff
@@ -0,0 +1,69 @@
+import Pikaday, { PikadayOptions } from 'pikaday';
+
+export interface AccessiblePikadayOptions extends PikadayOptions {
+    accessibleField?: HTMLElement | null;
+    datePickerButtonEl?: HTMLElement | null;
+}
+
+// An extended version of Pikaday to work when `isAccessible` prop is `true`. https://jira.inside-box.net/browse/A11Y-213
+class AccessiblePikaday extends Pikaday {
+    accessibleField: HTMLElement | null | undefined;
+
+    datePickerButtonEl: HTMLElement | null | undefined;
+
+    constructor(options: AccessiblePikadayOptions) {
+        super(options);
+        this.accessibleField = options.accessibleField;
+        this.datePickerButtonEl = options.datePickerButtonEl;
+
+        // Override behavior as if `options.field` and `options.bound` were set.
```

Comment:
Hm not sure about this one, @viodragon2 what do you think?

Reply:
> do we need to do any cleanup if `options.field` and `options.bound` were actually set?

Probably not, but it would be safe to do it in the component. When `AccessiblePikaday` is used in `DatePicker` https://github.com/box/box-ui-elements/pull/2751/files#diff-8c22974687f38296f0e4c9b61ba406c1a06dd3be54e61892b6a9ad5f546d0478R240 `field` property is deleted before initialized.

For `bound`, I'm not sure why it changed from my original change (https://github.com/box/box-ui-elements/pull/2733/files#diff-9b4042368d8ad5fdcc96cb31f69332780a5693527fb78ec9fbb8737e5af563bfR21-R23), but it seems like we're not overriding the behavior anymore for this case.

> Is it possible that this.el is appended to the document.body twice?

I don't think so? `this.el` is technically a private member of the `Pikaday` class and gets instantiated in its constructor.

Reply:
@viodragon2 I removed the click & focus listeners from your original change because the calendar popup should only open by clicking on the calendar icon. Focusing & clicking on the text box should not open the calendar.